keps/sig-instrumentation/1933-secret-logging-static-analysis/README.md

```diff
@@ -0,0 +1,677 @@
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+
+* **What are other known failure modes?**
+  For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+
+[supported limits]: https://git.k8s.io/community//sig-scalability/configs-and-limits/thresholds.md
+[existing SLIs/SLOs]: https://git.k8s.io/community/sig-scalability/slos/slos.md#kubernetes-slisslos
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+[GitHub's CodeQL](https://securitylab.github.com/tools/codeql) includes taint analysis
+and permits general SSA graph queries.  I am currently investigating if it is viable for this usecase.
+If so, integration with CodeQL / lgtm.com may be better than embedding analysis into the Prow testing.
+
+While other static analysis tools exist for Go, these tend towards more general linters.
+[`gosec`](https://github.com/securego/gosec), for instance, can be used to detect
+hard-coded tokens or use of cryptographically broken packages, e.g., `crypto/md5`.
+However, such linters are insufficient for our use, as they do not allow for project-specific configuration
+to identify sources, sinks, etc.
+
+There are few other developer analyzers that provide depth greater than a linter.
+[`gotcha`](https://github.com/akwick/gotcha), the <ins>go</ins> <ins>t</ins>aint <ins>ch</ins>ecker <ins>a</ins>nalyzer,
+provides a basic taint propagation analysis.
+However, it does not provide the ability to specify sanitizers or other fine-tuning.
```

Comment:
For metric-static-analysis, we hooked into the hack/verify_<blah> scripts, since it is the standard entrypoint. One downside is that it is getting overloaded these days, we should probably parallelize some of it.